src/core/components/inputs/TextInput.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {useEffect, useState} from 'react';
+import {StyledEyeBtn, StyledInputRowWrapper, StyledInputWrapper, StyledTextInput} from './Inputs.styled';
+import {
+    Keyboard,
+    KeyboardTypeOptions,
+    StyleProp,
+    TextStyle,
+    ViewStyle
+} from "react-native";
+import Ionicons from "@expo/vector-icons/Ionicons";
+import {RegularText} from "../textComponents/TextComponents";
+
+interface InputProps {
+    value: string,
+    required?: boolean,
+    keyboardType?: KeyboardTypeOptions,
+    autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters',
+    isEditable?: boolean,
+    multiline?: boolean,
+    maxLength?: number,
+    onChange: (arg0: string) => void,
+    errorHandler?: (arg0: boolean) => void,
+    label?: string,
+    placeholder?: string,
+    isPassword?: boolean,
+    validation?: { rule: RegExp, message: string }[],
+    styles?: {
+        wrapper?: StyleProp<ViewStyle>,
+        input?: StyleProp<TextStyle>,
+        label?: StyleProp<TextStyle>,
+        error?: StyleProp<TextStyle>
+    }
+}
+
+const TextInput: React.FC<InputProps> = (props) => {
+    const {
+        required = false,
+        value: initialValue,
+        keyboardType,
+        autoCapitalize = 'none',
+        isEditable = true,
+        multiline = false,
+        maxLength,
+        placeholder,
+        errorHandler,
+        isPassword,
+        validation,
+        onChange,
+        styles,
+        label
+    } = props;
+
+    const [value, setValue] = useState(initialValue)
+    const [error, setError] = useState('')
+    const [editable, setEditable] = useState(isEditable)
+    const [isFocused, setFocus] = useState(false)
+    const [secureText, setSecureText] = useState(isPassword);
+
+    useEffect(() => {
+        if (required && !value) {
+            errorHandler && errorHandler(true)
+        }
+    }, [])
+
+    // TODO: ###CF### Will produce extra re-render? Check and create custom useDidUpdate to get rid of it.
+    useEffect(() => {
+        setValue(initialValue);
+    }, [initialValue]);
+
+    useEffect(() => {
+        setEditable(isEditable);
+    }, [isEditable]);
+
+    const handlePasswordVisibilityChange = () => {
+        setSecureText(!secureText)
+    }
+
+    const handleChangeText = (value: string) => {
+        setValue(value);
+        if (validation?.length) {
+            const error = validation.find(el => !el.rule.test(value));
+            setError(!!error ? error.message : '');
+            errorHandler && errorHandler(!!error ? true : (required && !value));
+        } else {
+            errorHandler && errorHandler((required && !value));
+        }
+        onChange(value);
+    }
+
+    const handleFocus = () => {
+        setFocus(true);
+    }
+
+    const handleBlur = () => {
+        setFocus(false);
+        Keyboard.dismiss;
+    }
+
+
+
+    return (
```

Comment:
Please, delete the double empty rows? keep your code clean ))

Reply:
I think I need to add it as rule to eslint, cause sometimes I am inattentive to such small things. Thanks a lot!